### PR TITLE
Create a test snap for ethers.js compatibility check

### PR DESCRIPTION
**Pull request by bowensanders**
see issue #166

### CODE REVIEW

File: packages/site/src/features/ethersjs-snap/GetEntropy.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { FunctionComponent } from 'react';
+import { Snap } from '../../components';
+import { SignMessage } from './SignMessage';
+
+export const ETHERSJS_SNAP_ID = 'npm:@metamask/test-snap-ethersjs';
+export const ETHERSJS_PORT = 8014;
+
+export const Ethersjs: FunctionComponent = () => {
```
Comment (FrederikBolding) [inline packages/site/src/features/ethersjs-snap/GetEntropy.tsx:8]: Rename this file

File: packages/site/src/features/ethersjs-snap/SignMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          id="entropyMessage"
+          className="mb-3"
+        />
+
+        <Button type="submit" id="signEntropyMessage" disabled={isLoading}>
+          Sign Message
+        </Button>
+      </Form>
+      <Result>
+        <span id="entropySignResult">{JSON.stringify(data, null, 2)}</span>
```
Comment (FrederikBolding) [inline packages/site/src/features/ethersjs-snap/SignMessage.tsx:44]: This ID is already in use IIRC

File: packages/ethersjs/images/icon.svg
Comment (FrederikBolding) [inline packages/ethersjs/images/icon.svg:1]: I thought we were using the other icon now? 😅

File: packages/ethersjs/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# snap_ethersjs test Snap
```
Comment (FrederikBolding) [inline packages/ethersjs/README.md:1]:
```suggestion
# Ethersjs test Snap
```

File: packages/ethersjs/package.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    "registry": "https://registry.npmjs.org/"
+  },
+  "license": "ISC",
+  "files": [
+    "dist/",
+    "images/icon.svg",
+    "snap.manifest.json"
+  ],
+  "scripts": {
+    "build": "mm-snap build --eval false",
```
Comment (FrederikBolding) [inline packages/ethersjs/package.json:22]:
```suggestion
    "build": "mm-snap build --eval false",
```
Does it fail without `--eval false`?

File: packages/ethersjs/README.md (hunk 3 above)
Comment (bowensanders) [reply packages/ethersjs/README.md]: Fixed

File: packages/site/src/features/ethersjs-snap/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export * from './GetEntropy';
```
Comment (FrederikBolding) [inline packages/site/src/features/ethersjs-snap/index.ts:1]:
```suggestion
export * from './Ethersjs';
```

File: packages/ethersjs/package.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    "eth-rpc-errors": "^4.0.3",
+    "ethers": "^6.4.0"
+  },
+  "devDependencies": {
+    "@metamask/auto-changelog": "^2.5.0",
+    "@metamask/eslint-config": "^6.0.0",
+    "@metamask/eslint-config-jest": "^6.0.0",
+    "@metamask/eslint-config-nodejs": "^6.0.0",
+    "@metamask/eslint-config-typescript": "^6.0.0",
+    "@metamask/snaps-cli": "^0.30.0",
```
Comment (FrederikBolding) [inline packages/ethersjs/package.json:50]: Does this actually work? I thought transpiling might be broken on this version

File: packages/site/src/features/ethersjs-snap/SignMessage.tsx (hunk 2 above)
Comment (bowensanders) [reply packages/site/src/features/ethersjs-snap/SignMessage.tsx]: renamed

File: packages/ethersjs/package.json (hunk 4 above)
Comment (bowensanders) [reply packages/ethersjs/package.json:22]: This is the same in *all* of the package.json files in test-snaps... should it not be?

Comment (bowensanders) [reply packages/ethersjs/package.json]: I removed it and it still works

**Comment by FrederikBolding** [comment]
@SocketSecurity ignore aes-js@4.0.0-beta.5
@SocketSecurity ignore ethers@6.4.1

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**